source/frontend/src/features/leases/detail/LeasePages/details/DetailFeeDetermination.test.tsx

```diff
@@ -0,0 +1,139 @@
+import { Formik } from 'formik';
+import { createMemoryHistory } from 'history';
+import noop from 'lodash/noop';
+
+import { mockApiProperty } from '@/mocks/filterData.mock';
+import { getEmptyPropertyLease } from '@/mocks/properties.mock';
+import { ApiGen_Concepts_Lease } from '@/models/api/generated/ApiGen_Concepts_Lease';
+import { getEmptyLease } from '@/models/defaultInitializers';
+import { toTypeCode } from '@/utils/formUtils';
+import { render, RenderOptions } from '@/utils/test-utils';
+
+import { DetailFeeDetermination, IDetailFeeDeterminationProps } from './DetailFeeDetermination';
+
+const history = createMemoryHistory();
+
+describe('DetailFeeDetermination component', () => {
+  const setup = (
+    renderOptions: RenderOptions &
+      IDetailFeeDeterminationProps & { lease?: ApiGen_Concepts_Lease } = {},
+  ) => {
+    // render component under test
+    const component = render(
+      <Formik onSubmit={noop} initialValues={renderOptions.lease ?? getEmptyLease()}>
+        <DetailFeeDetermination
+          disabled={renderOptions.disabled}
+          nameSpace={renderOptions.nameSpace}
+        />
+      </Formik>,
+      {
+        ...renderOptions,
+        history,
+      },
+    );
+
+    return {
+      component,
+    };
+  };
+  it('renders minimally as expected', () => {
+    const { component } = setup({
+      lease: {
+        ...getEmptyLease(),
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...mockApiProperty,
+              areaUnit: toTypeCode('test'),
+              landArea: 123,
+            },
+            fileId: 1,
+          },
+        ],
+      },
+    });
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('renders a complete lease as expected', () => {
+    const { component } = setup({
+      lease: {
+        ...getEmptyLease(),
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...mockApiProperty,
+              areaUnit: toTypeCode('test'),
+              landArea: 123,
+            },
+          },
+        ],
+        amount: 1,
```

Comment:
when writing tests, please try and keep the test data as minimal as "focused" as possible on the parameters necessary to evaluate the component under test. For example, for this test, a large number of these fields will not have an effect on what is rendered, and can be removed.

Reply:
Reduced the amount of variable required for unit testing.